design/token-mechanics-v2.md

```diff
@@ -0,0 +1,324 @@
+# Token Mechanics V2
+
+### Assumptions
+- Users will register/deprecate versions of a package. 
+- Users will vouch/unvouch for an specific version of a package. 
+- Users will be able to move their vouch through different versions of the same package. 
+- Challengers will challenge a specific version of a package.
+- The minimum stake is validated against the summation of the amount vouched by the package owner for each of its version.
+- There is no penalization for unvouching/moving tokens
+- Packages reputation could be calculated based on total vouch amount weighted by the registered versions
+
+### Issues
+1- How we can avoid users unvouching their tokens given a new critical challenge yet unsolved?
+> We can either use cyclic windows to state when users can challenge, when users can vouch/unvouch, and when challenges 
+  will be solved. Or, we can lock a fraction of the vouched tokens of each user to ensure we can pay successful 
+  challenges. Both options have their downsides: locking may not be possible if there is a large list of challenges 
+  if the payout formula depends on the total stake. OTOH, windows may no be the ideal solution for critical challenges
+
+2- Challenges front-running?
+> We won't tackle this issue for this version.
+
+3- Challenges payout formula
+  3.1- Severity
+  > Each challenge can have a severity label given by the challenger. 
+
+  3.2- How we can make sure challenges payout cannot be manipulated based on their resolution order? 
+  > An obvious approach would be to solve challenges in a FIFO order.
+
+  3.3- How we can make sure there is enough tokens to pay successful challenges?
+  > We should decide whether we want to accept defaults or not.
```

Comment:
_Default_ may not be the best word in this context to explain this. Maybe "we want to accept Argentina-2001s"?